pilot/pkg/model/push_context.go

```diff
@@ -1136,13 +1137,14 @@ func (ps *PushContext) EnvoyFilters(proxy *Proxy) []*EnvoyFilterWrapper {
 
 // pre computes gateways per namespace
 func (ps *PushContext) initGateways(env *Environment) error {
-	gatewayConfigs, err := env.List(schemas.Gateway.Type, NamespaceAll)
+	gatewayConfigs, err := env.IstioConfigStore.List(schemas.Gateway.Type, NamespaceAll)
```

Comment:
what is the impact of this change?

Reply:
aren't they the same? One of the tests was segfaulting if the istioConfigStore was not used. So I thought it was something to do with how the fake store is setup.

Reply:
Not sure.. probably something that can/should be cleaned up. Seems most places we just use `env.List`